src/UIAutomation/UiaOperationAbstraction/UiaTypeAbstractionImpl.g.cpp

```diff
@@ -5020,6 +5030,93 @@
         return localPropertyValue;
     }
 
+    UiaCacheRequest::UiaCacheRequest(_In_ IUIAutomationCacheRequest* cacheRequest):
+        UiaTypeBase(MakeWinrtComPtr(cacheRequest))
+    {
+    }
+
+    UiaCacheRequest::UiaCacheRequest(winrt::com_ptr<IUIAutomationCacheRequest> const& cacheRequest):
+        UiaTypeBase(cacheRequest)
+    {
+    }
+

```

Comment:
For my understanding, I think the discussion around importing cache requests ended up on not needing that operation. And due to that, there is no `ToRemove()` call (and no supporting infrastructure to convert to remote).

From the practical point of view -- unless I am missing something -- this means that we should switch to using this (abstraction) type instead of `IUIAutomationCacheRequest` whenever we work with `UiaOperationAbstraction` (regardless of whether operations are executed locally or remotely) as otherwise:
1. The cache request will always be seen as empty/null.
2. Adding anything to the underlying cache request (when in the "remote" mode) will result in exceptions due to `std::get` finding out that the type of the cache request is not remote.

Is that correct to say? Or is there something I am missing? I am trying to understand the shift in paradigm of usability.

Previously, whomever used the abstraction did not really have to think twice about how abstraction types have been created. This sounds different, so I thought I would ask to understand the piece better. :)

Reply:
See above thread for more details, but yes, you're correct :). We shouldn't allow constructing using IUIAutomationCacheRequest.